features/other_b/bookmark_indexing.feature

```diff
@@ -60,48 +60,45 @@ Feature: Bookmark Indexing
     Then I should not see "Series"
 
   @new-search
-  Scenario: Synning a canonical tag used on a bookmarked series should move the 
-  bookmark to the new canonical's bookmark listings; de-synning should remove it
-    Given I am logged in as "author"
-      And a canonical fandom "Veronica Mars"
+  Scenario: Synning a canonical tag used on bookmarked series and external works
+  should move the bookmarks to the new canonical's bookmark listings; de-synning
+  should remove them
+    Given a canonical fandom "Veronica Mars"
       And a canonical fandom "Veronica Mars (TV)"
-      And I post the work "The Story Telling: Beginnings" with fandom "Veronica Mars" as part of a series "Telling Stories"
-      And I am logged in as "bookmarker"
-      And I bookmark the series "Telling Stories"
-    When I go to the bookmarks tagged "Veronica Mars"
-    Then the 1st bookmark result should contain "Telling Stories"
+      And bookmarks of external works and series tagged with the fandom tag "Veronica Mars"
     When I am logged in as a tag wrangler
       And I syn the tag "Veronica Mars" to "Veronica Mars (TV)"
       And I go to the bookmarks tagged "Veronica Mars (TV)"
-    Then the 1st bookmark result should contain "Telling Stories"
+    Then I should see "External Work to Bookmark"
+      And I should see "Series to Bookmark"
     When I de-syn the tag "Veronica Mars" from "Veronica Mars (TV)"
       And the tag "Veronica Mars" is canonized
       And I go to the bookmarks tagged "Veronica Mars (TV)"
-    Then I should not see "Telling Stories"
+    Then I should not see "External Work to Bookmark"
+      And I should not see "Series to Bookmark"
     When I go to the bookmarks tagged "Veronica Mars"
-    Then the 1st bookmark result should contain "Telling Stories"
+    Then I should see "External Work to Bookmark"
```

Comment:
The fact that this step works is somewhat deceptive. At this point, the ExternalWork in the Veronica Mars tag doesn't have any FilterTaggings related to the Veronica Mars tag, the way that it ought to. (This is because `add_filter_taggings`, which is the only thing that creates FilterTaggings when a tag goes from non-canonical to canonical, currently doesn't process ExternalWorks.) The only reason that this ExternalWork shows up in the Veronica Mars tag is because it's directly tagged with "Veronica Mars," and the filter_ids field for all Taggable items is defined to be `(tags.pluck(:id) + filters.pluck(:id)).uniq`.

If you attempted to syn Veronica Mars to "Veronica Mars (TV)" again after this step, the ExternalWork would not show up in the "Veronica Mars (TV)" tag, because `add_filter_taggings` wouldn't add the new filter, and `move_filter_taggings_to_merger` would have no FilterTaggings to transfer.

Reply:
Yeah. The issue with `add_filter_taggings` had a separate Jira issue and should be fixed by the next PR, #3294.